packages/aws-cdk-lib/aws-fsx/lib/lustre-file-system.ts

```diff
@@ -384,16 +407,23 @@ export class LustreFileSystem extends FileSystemBase {
   /**
    * Validates the perUnitStorageThroughput is defined correctly for the given deploymentType.
    */
-  private validatePerUnitStorageThroughput(deploymentType: LustreDeploymentType, perUnitStorageThroughput?: number) {
+  private validatePerUnitStorageThroughput(
+    deploymentType: LustreDeploymentType,
+    perUnitStorageThroughput?: number,
+    storageType?: StorageType,
+  ): void {
     if (perUnitStorageThroughput === undefined) { return; }
 
     if (deploymentType !== LustreDeploymentType.PERSISTENT_1 && deploymentType !== LustreDeploymentType.PERSISTENT_2) {
       throw new Error('perUnitStorageThroughput can only be set for the PERSISTENT_1/PERSISTENT_2 deployment types, received: ' + deploymentType);
     }
 
     if (deploymentType === LustreDeploymentType.PERSISTENT_1) {
-      if (![50, 100, 200].includes(perUnitStorageThroughput)) {
-        throw new Error('perUnitStorageThroughput must be 50, 100, or 200 MB/s/TiB for PERSISTENT_1 deployment type, got: ' + perUnitStorageThroughput);
+      if (storageType === StorageType.HDD && ![12, 40].includes(perUnitStorageThroughput)) {
+        throw new Error(`perUnitStorageThroughput must be 12 or 40 MB/s/TiB for PERSISTENT_1 HDD storage, got: ${perUnitStorageThroughput}`);
+      }
+      if ((storageType === undefined || storageType === StorageType.SSD) && ![50, 100, 200].includes(perUnitStorageThroughput)) {
+        throw new Error('perUnitStorageThroughput must be 50, 100, or 200 MB/s/TiB for PERSISTENT_1 SSD storage, got: ' + perUnitStorageThroughput);
       }
```

Comment:
Can you add validation for `PERSISTENT_2` as well ([docs](https://docs.aws.amazon.com/fsx/latest/LustreGuide/managing-throughput-capacity.html))?

Reply:
I've added!

```ts
    if (deploymentType === LustreDeploymentType.PERSISTENT_2) {
      if (![125, 250, 500, 1000].includes(perUnitStorageThroughput)) {
        throw new Error(`perUnitStorageThroughput must be 125, 250, 500 or 1000 MB/s/TiB for PERSISTENT_2 deployment type, got: ${perUnitStorageThroughput}`);
      }
    }
```